src/main/java/seedu/exercise/model/book/ReadOnlyResourceBook.java

```diff
@@ -0,0 +1,98 @@
+package seedu.exercise.model.book;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.exercise.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.exercise.model.Resource;
+import seedu.exercise.model.UniqueList;
+
+/**
+ * Encapsulates a Resource Book that can contain {@code Resource} objects of type {@code T}.
+ */
+public class ReadOnlyResourceBook<T extends Resource> {
+
+    protected final UniqueList<T> resources;
```

Comment:
Why this is protected? Do subclasses need it? Can make it private and make a getter for it

Reply:
Changed to private and made a getter for it.